app/models/manageiq/providers/ibm_power_hmc/inventory/collector/infra_manager.rb

```diff
@@ -57,8 +59,30 @@ def templates
     @templates || []
   end
 
+  attr_accessor :ssps
+
+  def ssp_lus_by_udid
+    @ssp_lus_by_udid ||= {}
+    ssps.each { |ssp| ssp.lus.each { |lu| @ssp_lus_by_udid[lu.udid] = ssp.cluster_uuid } }
+    @ssp_lus_by_udid
```

Comment:
So the way this is written it will redefine the `@ssp_lus_by_udid` everytime this is called.

You probably want something like:
```
@ssp_lus_by_udid ||= ssps.each_with_object({}) { |ssp, result| ssp.lus.each { |lu| result[lu.udid] = ssp.cluster_uuid } }
```

There's probably a cleaner way to one-line that but it is escaping me at the moment

Reply:
There is also the begin/end way of doing this (if a one-liner does not work for some reason):

```ruby
@ssp_lus_by_udid ||= begin
  hash = {}
  ssps.each { |ssp| ssp.lus.each { |lu| hash[lu.udid] = ssp.cluster_uuid } }
  hash
end
```

Having said that, I think the one-liner is clearer and more appropriate in this case.

Reply:
Something like `@ssp_lus_by_udid ||= ssps.flat_map { |ssp| ssp.lus.map { |lu| [lu.udid, ssp.cluster_uuid] } }.to_h` should do the trick also